CMakeLists.txt

```diff
@@ -745,10 +746,16 @@ endif (WITH_PKGCONFIG)
 # Unit testing
 
 if (WITH_FUZZING AND NOT WITH_FUZZING STREQUAL "none")
-  add_executable (fuzz_demangle
+  add_executable (fuzz_detangle
     src/fuzz_demangle.cc
   )
-  target_link_libraries (fuzz_demangle PRIVATE glog ${WITH_FUZZING})
+  if (WITH_FUZZING STREQUAL "ossfuzz")
+    target_link_libraries (fuzz_demangle PRIVATE glog $ENV{LIB_FUZZING_ENGINE})
```

Comment:
Why do we need `LIB_FUZZING_ENGINE` here? Isn't the `WITH_FUZZING` value sufficient? Either way, I would avoid using an environment variable here and instead use a CMake variable directly (drop `$ENV`). One can always initialize a CMake variable using an environment variable which is the usual pattern.

Reply:
oss-fuzz uses this environment variable to use whatever engine such as libfuzzer, afl++ or honggfuzz or other ones...

As you said 

> The option is not meant to hold compiler flags or the like. It should be a symbolic name (e.g., fuzzer)

So, I chose ossfuzz for this setup.
What do you think about it ?

So, I divide in 2 lines : setting the cmake variable from the environment and then using it...

Reply:
Shouldn't we split `ossfuzz` into libfuzzer, afl++ or honggfuzz then? This would provide a much better overview of the choices we have.

Reply:
I do not think so because oss-fuzz does not use regular afl++

And oss-fuzz is adding more engines, you just keep the env variable `LIB_FUZZING_ENGINE`, and get these for free

Reply:
I see. Let's then keep it this way.